ztools/x509/json.go

```diff
@@ -220,7 +221,16 @@ func (c *Certificate) MarshalJSON() ([]byte, error) {
 	jc.Subject = c.Subject
 	jc.SubjectDN = c.Subject.String()
 	jc.SubjectKeyInfo.KeyAlgorithm = c.PublicKeyAlgorithm
-	jc.SubjectKeyInfo.SPKIFingerprint = c.SPKIFingerprint
+
+	// 1 for the DN, and then however many DNS names there are
+	jc.Names = make([]string, 1+len(c.DNSNames))
+
+	jc.Names[0] = jc.SubjectDN
```

Comment:
The `SubjectDN` isn't a URL. I think you'll need to go through each of `[]Names` in `pkix.Name`, check to see if it's a URL, and add it if it is. Golang's `net.URL` probably helps with this.